sdk/core/Azure.Core.Experimental/src/Primitives/BinaryData.cs

```diff
@@ -73,6 +74,24 @@ public BinaryData(string data)
             Bytes = s_encoding.GetBytes(data);
         }
 
+        /// <summary>
+        /// Creates a binary data instance from an object and serializes it
+        /// using the provided <see cref="ObjectSerializer"/>. If no <see cref="ObjectSerializer"/>
+        /// is specified, <see cref="JsonObjectSerializer"/> will be used.
+        /// </summary>
+        /// <param name="data">The data that will be serialized.</param>
+        /// <param name="type">The type of the data.</param>
+        /// <param name="serializer">The serializer to serialize
+        /// the data.</param>
+        /// <returns>A <see cref="BinaryData"/> instance.</returns>
+        public BinaryData(object data, Type type, ObjectSerializer? serializer = default)
+        {
+            using var memoryStream = new MemoryStream();
+            serializer ??= new JsonObjectSerializer();
```

Comment:
File a bug to make this less allocaling

Reply:
https://github.com/Azure/azure-sdk-for-net/issues/14346